### DISCUSSION TITLE
Modifying the source code to get F_ext as an output

### DISCUSSION THREAD

**Discussion by relhamoud** [OpenFAST/openfast#2446] [Q&A] [answered]
Hello everyone,

I have a question regarding the modification of the source code of OpenFAST in order to add  a new output channel called F_ext. This variable existed initially in OpenFAST but i want to extract its value at each time step. 
Could you please help me and tell me from where i should start ?

Best Regards,

Riad

**Comment by jjonkman** [comment]
Dear @relhamoud,

I would suggest reviewing the source code for how other ServoDyn Structural Control (StC)-related outputs are implemented (e.g., `NStC1_XQ`) and follow suite.

Note that some of the code can be auto-generated after changing _OutListParameters.xlsx_ using the provided _Write_ChckOutList.m_ script.

Best regards,

**Reply by relhamoud** [reply]
Dear @jjonkman 

I have modified the file named "ServoDyn_IO.f90". Is it the right file to modify ? and is there any other file to modify ?

Best Regards,

Riad

**Reply by jjonkman** [reply, marked as answer]
_ServoDyn_IO.f90_ is the only file you should need to modify to add other ServoDyn Structural Control (StC)-related outputs.

Best regards,
